src/cdk-experimental/listbox/listbox.ts

```diff
@@ -8,182 +8,146 @@
   },
 })
-export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightable {
-  private _selected: boolean = false;
-  private _disabled: boolean = false;
-  private _value: T;
-  _active: boolean = false;
+export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightable, OnDestroy {
+  /** The id of the option's host element. */
+  @Input() id = `cdk-option-${nextOptionId++}`;
 
-  /** The id of the option, set to a uniqueid if the user does not provide one. */
-  @Input() id = `cdk-option-${nextId++}`;
+  // TODO(mmalerba): What do we do if the user doesn't specify a value?
+  /** The value of this option. */
+  @Input('cdkOptionValue') value: T;
 
-  @Input()
-  get selected(): boolean {
-    return this._selected;
-  }
-  set selected(value: BooleanInput) {
-    if (!this._disabled) {
-      this._selected = coerceBooleanProperty(value);
-    }
-  }
+  /**
+   * The text used to locate this item during listbox typeahead. If not specified,
+   * the `textContent` of the item will be used.
+   */
+  @Input('cdkOptionTypeaheadLabel') typeaheadLabel: string;
 
-  @Input()
+  /** Whether this option is disabled. */
+  @Input('cdkOptionDisabled')
   get disabled(): boolean {
-    return this._disabled;
+    return this.listbox.disabled || this._disabled;
   }
   set disabled(value: BooleanInput) {
     this._disabled = coerceBooleanProperty(value);
   }
+  private _disabled: boolean = false;
 
-  /** The form value of the option. */
-  @Input()
-  get value(): T {
-    return this._value;
-  }
-  set value(value: T) {
-    if (this.selected && value !== this._value) {
-      this.deselect();
-    }
-    this._value = value;
-  }
+  /** Emits when the option is destroyed. */
+  protected destroyed = new Subject<void>();
 
-  /**
-   * The text used to locate this item during menu typeahead. If not specified,
-   * the `textContent` of the item will be used.
-   */
-  @Input() typeahead: string;
+  /** Emits when the option is clicked. */
+  readonly _clicked = new Subject<void>();
 
-  @Output() readonly selectionChange = new EventEmitter<OptionSelectionChangeEvent<T>>();
+  /** Whether the option is currently active. */
+  private _active = false;
 
   constructor(
-    private readonly _elementRef: ElementRef,
-    @Inject(forwardRef(() => CdkListbox)) readonly listbox: CdkListbox<T>,
-  ) {}
-
-  /** Toggles the selected state, emits a change event through the injected listbox. */
-  toggle() {
-    if (!this._isInteractionDisabled()) {
-      this.selected = !this.selected;
-      this._emitSelectionChange(true);
-    }
-  }
-
-  /** Sets the active property true if the option and listbox aren't disabled. */
-  activate() {
-    if (!this._isInteractionDisabled()) {
-      this._active = true;
-    }
+    /** The option's host element */
+    protected readonly elementRef: ElementRef,
+    /** The change detector ref for this option. */
+    protected readonly changeDetectorRef: ChangeDetectorRef,
```

Comment:
Excellent news- as of https://github.com/angular/angular/pull/45991, we no longer have to exclusively use constructor injection. Now we should be able to do e.g.:

```ts
import {inject} from '@angular/core';
...
export class CdkOption {
  private changeDetectorRef = inject(ChangeDetectorRef);
  private nativeElement = inject(ElementRef).nativeElement;
}
```

Reply:
Actually, this PR has been reverted for the moment

Reply:
Double actually, that revert never got merged, original comment stands